backend/push.js

```diff
@@ -242,21 +198,19 @@ export const postEvent = async (subscription: Object, event: ?string): Promise<v
   })
 
   if (!req.ok) {
-    console.warn('Error sending push notification', subscription.id, req.status)
+    const endpointHost = new URL(subscription.endpoint).host
+    console.warn(`Error ${req.status} sending push notification to '${subscription.id}' via ${endpointHost}`)
     // If the response was 401 (Unauthorized), 404 (Not found) or 410 (Gone),
     // it likely means that the subscription no longer exists.
     if ([401, 404, 410].includes(req.status)) {
-      console.warn(
-        new Date().toISOString(),
-        'Removing subscription',
-        subscription.id
-      )
-      deleteClient(subscription.id)
-      return
+      console.warn(new Date().toISOString(), 'Removing subscription', subscription.id)
+      await removeSubscription(subscription.id)
```

Comment:
By adding these various `await` statements, you're slowing things down, such as broadcasting messages.

Reply:
I can remove this one, sure. The `await`s are important though, as without them the code is difficult to reason about and that leads to bugs.